retries.go

```diff
@@ -54,7 +57,19 @@ func tooManyRequestsRetryCondition(r *resty.Response, _ error) bool {
 }
 
 func serviceUnavailableRetryCondition(r *resty.Response, _ error) bool {
-	return r.StatusCode() == http.StatusServiceUnavailable
+	var retry bool
+	if r.StatusCode() == http.StatusServiceUnavailable {
+		// During maintenance events, the API will return a 503 and add
+		// an `X-MAINTENANCE-MODE` header. Don't rety during maintenance
+		// events, only for legitimate 503s.
+		if r.Header().Get(maintenanceModeHeaderName) != "" {
+			log.Printf("[INFO] Linode API is under maintenance, request will not be retried")
```

Comment:
Should we also log a message telling users where they can go for more information? Ie, add something like

> Please see status.linode.com for more details

🤔

Reply:
I like this idea a lot! Feel free to add it

Reply:
Added in 6d36a3d069c72b413f6741fdf26fbadfc3245065 👍